samples/net/udp/boards/native_sim.conf

```diff
@@ -0,0 +1,56 @@
+#
+# Copyright (c) 2023 Nordic Semiconductor ASA
+#
+# SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+#
+
+# NEWLIB C
+CONFIG_NEWLIB_LIBC=n
+CONFIG_NEWLIB_LIBC_FLOAT_PRINTF=n
+
+# Networking config
+CONFIG_NET_IPV6=y
+CONFIG_NET_TCP=y
+CONFIG_NET_UDP=y
+CONFIG_NET_SOCKETS_POSIX_NAMES=y
+CONFIG_NET_SOCKETS_SOCKOPT_TLS=y
+CONFIG_NET_PKT_RX_COUNT=32
+CONFIG_NET_PKT_TX_COUNT=32
+CONFIG_NET_BUF_RX_COUNT=32
+CONFIG_NET_BUF_TX_COUNT=32
+CONFIG_NET_IF_UNICAST_IPV6_ADDR_COUNT=3
+CONFIG_NET_IF_MCAST_IPV6_ADDR_COUNT=4
+CONFIG_NET_MAX_CONTEXTS=10
+CONFIG_NET_L2_ETHERNET=y
+CONFIG_NET_CONTEXT_SNDTIMEO=y
+
+CONFIG_ETH_NATIVE_POSIX=y
+CONFIG_ETH_NATIVE_POSIX_RANDOM_MAC=y
+
+# Network driver config
+CONFIG_TEST_RANDOM_GENERATOR=y
+
+# DNS
+CONFIG_DNS_RESOLVER=y
+CONFIG_DNS_SERVER_IP_ADDRESSES=y
+CONFIG_DNS_SERVER1="8.8.8.8"
+CONFIG_DNS_RESOLVER_MAX_SERVERS=2
+
+# Network address config
+CONFIG_NET_CONFIG_SETTINGS=y
+CONFIG_NET_CONFIG_NEED_IPV4=y
+CONFIG_NET_CONFIG_MY_IPV4_ADDR="192.0.2.1"
+CONFIG_NET_CONFIG_PEER_IPV4_ADDR="192.0.2.2"
+CONFIG_NET_CONFIG_MY_IPV4_GW="192.0.2.2"
+
+CONFIG_ENTROPY_GENERATOR=y
+CONFIG_NATIVE_POSIX_SLOWDOWN_TO_REAL_TIME=y
+
+
+# LTE link control
```

Comment:
This one and the once below shouldn't be necessary to disable since modem libraries are only pulled in when building for nrf91 based boards.